From d4e5d65c817b9306349fb7b3956af4f7e798ddfd Mon Sep 17 00:00:00 2001
From: Marius Bagassien <marius.b@variscite.com>
Date: Fri, 13 Mar 2026 10:00:39 +0100
Subject: [PATCH 1/2] variscite: bootanimation: Add build-time logo override
 via product_variables

Allow BSP vendors to substitute the android() boot animation logo
textures (mask and shine) at build time by setting:

  PRODUCT_BOOTANIMATION_LOGO_MASK
  PRODUCT_BOOTANIMATION_LOGO_SHINE

in their product Makefile. Falls back to the stock Google assets when
the variables are not set, preserving existing behavior.

Signed-off-by: Marius Bagassien <marius.b@variscite.com>
---
 cmds/bootanimation/Android.bp        | 32 +++++++++++++++++++++++++++-
 cmds/bootanimation/BootAnimation.cpp | 14 ++++++++++--
 2 files changed, 43 insertions(+), 3 deletions(-)

diff --git a/cmds/bootanimation/Android.bp b/cmds/bootanimation/Android.bp
index 3534624a58a2..41c5f780e9b7 100644
--- a/cmds/bootanimation/Android.bp
+++ b/cmds/bootanimation/Android.bp
@@ -1,3 +1,30 @@
+// Variscite BSP: build-time boot animation logo override.
+// Set PRODUCT_BOOTANIMATION_LOGO_MASK and PRODUCT_BOOTANIMATION_LOGO_SHINE
+// in your product Makefile to override the default Google logo textures.
+
+soong_config_module_type {
+    name: "bootanimation_logo_cc_defaults",
+    module_type: "cc_defaults",
+    config_namespace: "bootanimation",
+    value_variables: [
+        "logo_mask",
+        "logo_shine",
+    ],
+    properties: ["cflags"],
+}
+
+bootanimation_logo_cc_defaults {
+    name: "bootanimation_logo_defaults",
+    soong_config_variables: {
+        logo_mask: {
+            cflags: ["-DBOOT_LOGO_MASK=\"%s\""],
+        },
+        logo_shine: {
+            cflags: ["-DBOOT_LOGO_SHINE=\"%s\""],
+        },
+    },
+}
+
 package {
     // See: http://go/android-license-faq
     // A large-scale-change added 'default_applicable_licenses' to import
@@ -63,7 +90,10 @@ cc_binary {
 
 cc_library_shared {
     name: "libbootanimation",
-    defaults: ["bootanimation_defaults"],
+    defaults: [
+        "bootanimation_defaults",
+        "bootanimation_logo_defaults",
+    ],
 
     srcs: ["BootAnimation.cpp"],
 
diff --git a/cmds/bootanimation/BootAnimation.cpp b/cmds/bootanimation/BootAnimation.cpp
index c2f6e3072507..8b1aefc2fd10 100644
--- a/cmds/bootanimation/BootAnimation.cpp
+++ b/cmds/bootanimation/BootAnimation.cpp
@@ -108,6 +108,16 @@ static const char EXIT_PROP_NAME[] = "service.bootanim.exit";
 static const char PROGRESS_PROP_NAME[] = "service.bootanim.progress";
 static const char DISPLAYS_PROP_NAME[] = "persist.service.bootanim.displays";
 static const char CLOCK_ENABLED_PROP_NAME[] = "persist.sys.bootanim.clock.enabled";
+// Boot animation logo paths.
+// Override at build time by setting PRODUCT_BOOTANIMATION_LOGO_MASK and
+// PRODUCT_BOOTANIMATION_LOGO_SHINE in your product Makefile.
+#ifndef BOOT_LOGO_MASK
+#define BOOT_LOGO_MASK "images/android-logo-mask.png"
+#endif
+#ifndef BOOT_LOGO_SHINE
+#define BOOT_LOGO_SHINE "images/android-logo-shine.png"
+#endif
+
 static const int ANIM_ENTRY_NAME_MAX = ANIM_PATH_MAX + 1;
 static const int MAX_CHECK_EXIT_INTERVAL_US = 50000;
 static constexpr size_t TEXT_POS_LEN_MAX = 16;
@@ -875,8 +885,8 @@ bool BootAnimation::android() {
 
     SLOGD("%sAnimationShownTiming start time: %" PRId64 "ms", mShuttingDown ? "Shutdown" : "Boot",
             elapsedRealtime());
-    initTexture(&mAndroid[0], mAssets, "images/android-logo-mask.png");
-    initTexture(&mAndroid[1], mAssets, "images/android-logo-shine.png");
+    initTexture(&mAndroid[0], mAssets, BOOT_LOGO_MASK);
+    initTexture(&mAndroid[1], mAssets, BOOT_LOGO_SHINE);
 
     mCallbacks->init({});
 
-- 
2.43.0

